pkg/descheduler/strategies/lownodeutilization.go

```diff
@@ -311,10 +312,15 @@ func sortPodsBasedOnPriority(evictablePods []*v1.Pod) {
 }
 
 // createNodePodsMap returns nodepodsmap with evictable pods on node.
-func createNodePodsMap(client clientset.Interface, nodes []*v1.Node) NodePodsMap {
+func createNodePodsMap(client clientset.Interface, podSelector string, nodes []*v1.Node) NodePodsMap {
 	npm := NodePodsMap{}
+	selector, err := labels.Parse(podSelector)
+	if err != nil {
+		return npm
+	}
+
 	for _, node := range nodes {
-		pods, err := podutil.ListPodsOnANode(client, node)
+		pods, err := podutil.ListPodsOnANode(client, selector, node)
```

Comment:
pod selectors might not work right for lownodeutilization startgey this way as in this strategy, a node's load is computed by considering all pods. So computing loads on each node should be done by considering all pods. Perhaps during only eviction, pod selectors might be used.

Reply:
Would it be better if I moved the implementation of this into podutil.isEvictable rather than podutil.ListPodsOnNode? I considered this, but it means a label check on each pod which slows down the check as pod numbers go up. Would it work to make lownodeutilization a special case? I guess it already works quite a bit differently to the other strategies.